airflow/models/dag.py

```diff
@@ -734,15 +760,25 @@ def task_group(self) -> "TaskGroup":
 
     @property
     def filepath(self) -> str:
-        """File location of where the dag object is instantiated"""
-        fn = self.full_filepath.replace(settings.DAGS_FOLDER + '/', '')
-        fn = fn.replace(os.path.dirname(__file__) + '/', '')
-        return fn
+        """:meta private:"""
+        warnings.warn(
+            "filepath is deprecated, use relative_fileloc instead", DeprecationWarning, stacklevel=2
+        )
+        return self.relative_fileloc
+
+    @property
+    def relative_fileloc(self) -> str:
+        """File location of the importable dag 'file' relative to the configured DAGs folder."""
+        try:
+            return str(pathlib.Path(self.fileloc).relative_to(settings.DAGS_FOLDER))
```

Comment:
Why not just return the Path object? This is new so there's no compatibility considerations, and a Path can be used in almost all places where str is accepted these days (`os.PathLike`).

Reply:
Fair point -- this can return a path object, and the compat layers that call it can force a string return.